ios/browser/api/ledger/brave_ledger.mm

```diff
@@ -392,14 +372,13 @@ - (NSString*)rewardsDatabasePath {
 }
 
 - (void)resetRewardsDatabase {
-  delete rewardsDatabase;
```

Comment:
is this running on the main thread? If so this code was broken in terms of threading. If it's running on the task runner then it's not safe to initialize the new db here.

Reply:
if it's running on the main thread then the new code is ok

Reply:
@kylehickinson verified that this is run on the main thread and only in some debug mode so the new code is fine

Reply:
Yep always main thread, but also this code is only run in debug mode when resetting rewards (never run in production atm)